
# HG changeset patch
# User Christoph Moench-Tegeder <cmt@burggraben.net>
# Date 1712783081 0
# Node ID 15006906e885bda833a3f867cfd8ced955b3b5d5
# Parent  bc425d90bbe9987cc754323772553d4be3e6de83
Bug 1890593 - GetSystemWPADSetting for libproxy-enabled builds r=jesup

In #621429 a call to nsUnixSystemProxySettings::GetSystemWPADSetting()
was added, but that function is not available for builds with
MOZ_ENABLE_LIBPROXY. The funtion is trivial enough to just
copy it across - else there'd more boilerplate than actual code.

Differential Revision: https://phabricator.services.mozilla.com/D207197

diff --git a/toolkit/system/unixproxy/nsLibProxySettings.cpp b/toolkit/system/unixproxy/nsLibProxySettings.cpp
--- a/toolkit/system/unixproxy/nsLibProxySettings.cpp
+++ b/toolkit/system/unixproxy/nsLibProxySettings.cpp
@@ -94,11 +94,17 @@ nsresult nsUnixSystemProxySettings::GetP
 
     c++;
   }
 
   free(proxyArray);
   return NS_OK;
 }
 
+NS_IMETHODIMP
+nsUnixSystemProxySettings::GetSystemWPADSetting(bool* aSystemWPADSetting) {
+  *aSystemWPADSetting = false;
+  return NS_OK;
+}
+
 NS_IMPL_COMPONENT_FACTORY(nsUnixSystemProxySettings) {
   return do_AddRef(new nsUnixSystemProxySettings()).downcast<nsISupports>();
 }

